device/thunder_ripple_sdk/src/client/thunder_async_client.rs

```diff
@@ -0,0 +1,617 @@
+                device_response_msg = Some(DeviceResponseMessage::new(dev_resp, sub_id));
+            }
+        } else {
+            error!("deviceresponse msg extraction failed.");
+        }
+        println!("@@NNA...device_response_msg: {:?}", device_response_msg);
+        device_response_msg
+    }
+}
+
+impl ThunderAsyncClient {
+    fn get_id_from_result(result: &[u8]) -> Option<u64> {
+        serde_json::from_slice::<JsonRpcApiResponse>(result)
+            .ok()
+            .and_then(|data| data.id)
+    }
+
+    pub fn get_sender(&self) -> BrokerSender {
+        self.sender.clone()
+    }
+
+    pub fn get_thndrasync_callback(&self) -> BrokerCallback {
+        self.callback.clone()
+    }
+    async fn create_ws(
+        endpoint: &str,
+    ) -> (
+        SplitSink<WebSocketStream<TcpStream>, Message>,
+        SplitStream<WebSocketStream<TcpStream>>,
+    ) {
+        info!("Broker Endpoint url {}", endpoint);
+
+        let url = url::Url::parse(endpoint).unwrap();
+        let port = extract_tcp_port(endpoint);
+        info!("Url host str {}", url.host_str().unwrap());
+        let mut index = 0;
+
+        loop {
+            // Try connecting to the tcp port first
+            if let Ok(v) = TcpStream::connect(&port).await {
+                // Setup handshake for websocket with the tcp port
+                // Some WS servers lock on to the Port but not setup handshake till they are fully setup
+                if let Ok((stream, _)) = client_async(endpoint, v).await {
+                    break stream.split();
+                }
+            }
+            if (index % 10).eq(&0) {
+                error!(
+                    "Broker with {} failed with retry for last {} secs in {}",
+                    endpoint, index, port
+                );
+            }
+            index += 1;
+            tokio::time::sleep(Duration::from_secs(1)).await;
+        }
+    }
+
+    fn prepare_request(&self, request: &ThunderAsyncRequest) -> Result<Vec<String>, RippleError> {
+        println!("@@@NNA-----> in start prepare request id :{}", request.id);
+        let mut requests = Vec::new();
+        let id: u64 = request.id;
+        let (callsign, method) = request.request.get_callsign_method();
+        println!(
+            "@@@NNA-----> in prepare request callsign :{}, method :{}",
+            callsign, method
+        );
+        if method.is_empty() {
+            return Err(RippleError::InvalidInput);
+        }
+        // check if the plugin is activated.
+        let status = match self.status_manager.get_status(callsign.clone()) {
+            Some(v) => {
+                println!(
+                    "@@@NNA....status_manager.get_status returned some value. v:{:?} ",
+                    v
+                );
+                v.clone()
+            }
+            None => {
+                println!("@@@NNA....status_manager.get_status return none.");
+                self.status_manager
```

Comment:
I see that `ThunderAsyncClient` has it's own plug-in status manager, this doesn't seem right since we already have one in `thunder_plugins_status_mgr.rs`. Does the existing manager not provide something you need? If not, then instead of creating another we should look into refactoring the existing `StatusManager` so that it suits both new and existing needs. Let's discuss what is needed to do this.